src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/NotifyService.kt

```diff
@@ -1,32 +1,86 @@
 package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
 
+import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.context.ApplicationListener
 import org.springframework.stereotype.Service
 import org.springframework.web.util.UriComponentsBuilder
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.component.EmailSender
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ActionPlanEvent
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ActionPlanEventType
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ReferralEvent
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ReferralEventType
-import uk.gov.service.notify.NotificationClient
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
 import java.net.URI
 import java.util.UUID
 
+interface NotifyService {
+
+  fun generateResourceUrl(baseURL: String, path: String, id: UUID): URI {
+    return UriComponentsBuilder.fromHttpUrl(baseURL).path(path).buildAndExpand(id).toUri()
+  }
+
+  fun getUserDetails(user: AuthUser, hmppsAuthService: HMPPSAuthService, log: Logger) =
+    try {
+      hmppsAuthService.getUserDetail(user)
+    } catch (e: Exception) {
+      log.error("could not get account details for assigned service provider user", e)
+      null
```

Comment:
is this neccesary?

Reply:
are we missing a return for this method?

Reply:
Yeah the null is necessary as the method expects to return a UserDetail instance. Perhaps a clearer way of writing it could be

null as UserDetail